Option to require scrolling to the bottom of ORKConsentReviewStep

eschramm opened the pull request:
This adds an additional property to `ORKConsentReviewStep` called `requresScrollToBottom`.  When set to true, the "Agree" button will not be activated until the user scrolls all the way to thee bottom of the content.  This is for a particular research project we are working with where it's very important that the participant fully read and understand the consent - and we figured it would make sense to contribute back to ResearchKit in case someone else could use the feature.

![mustscroll](https://user-images.githubusercontent.com/1008462/42818157-b8e9946a-8995-11e8-8c5e-410837fc47b5.gif)

ninoguba commented:
Hi. This is pretty useful.  My only concern is backward compatibility since you've modified the init methods. I think it would be better if you just overload the init methods with your new requiresScrollToBottom parameter and default it to false if it is not being set in the init parameter.

eschramm commented:
I like that idea, thank you for the suggestion.  I have updated to restore the previous initializer as a convenience initializer so as not to change behavior for any existing code.

ninoguba commented on ResearchKit/Consent/ORKConsentReviewStep.h:
```diff
@@ -67,12 +67,30 @@ ORK_CLASS_AVAILABLE
 /**
  Returns an initialized consent review step using the specified identifier, signature, and consent document.
```
Please include here the new requiresScrollToBottom parameter for a more complete documentation.

srinathtm-apple commented:
Thanks for submitting this @eschramm , this is really useful.

Also, I was wondering.. wouldn't it be easier if we made this a property of ORKConsentReviewStep instead of adding a brand new initializer? The default behavior is that the property will be set to false and it will be easier to adopt.

@ninoguba - would appreciate your thoughts on this as well..

eschramm commented:
I have removed the new parameter from initializers, leaving it to be modified from false after task creation if desired.  This seems like a less invasive approach 👍

eschramm commented on ResearchKit/Consent/ORKConsentReviewController.m:
```diff
@@ -80,6 +84,7 @@ - (void)viewDidLoad {
     [_webView loadHTMLString:_htmlString baseURL:ORKCreateRandomBaseURL()];
     _webView.backgroundColor = ORKColor(ORKConsentBackgroundColorKey);
     _webView.scrollView.backgroundColor = ORKColor(ORKConsentBackgroundColorKey);
+    _webView.scrollView.delegate = self;
```
That makes sense, thanks for the suggestion.

eschramm commented:
Have updated to make the `UIScrollViewDelegate` conditional to minimize callbacks.  Also added support for an edge case we've encountered.  If a short consent document is loaded such that no scrolling is needed to reach the bottom, the Agree button is inexplicably disabled until a scroll attempt is performed.  This may confuse the user and because different device sizes and orientations may affect the need to scroll, we added a check after the web view renders and re-enables the Agree button if already at the bottom.

srinathtm-apple commented:
Merged into master, thanks a bunch @eschramm !
